blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MultibranchPipelineRunContainer.java

```diff
@@ -85,22 +83,26 @@ public BlueRun get(String name) {
             sortBranchesByLatestRun(branches);
         }
 
+        List<BlueRun> allRunList = new ArrayList<BlueRun>();
         for (final BluePipeline b : branches) {
             BlueRunContainer blueRunContainer = b.getRuns();
             if(blueRunContainer==null){
                 continue;
             }
-            Iterator<BlueRun> it = blueRunContainer.iterator(0, MAX_MBP_RUNS_ROWS);
-            int count = 0;
-            Utils.skip(it, start);
-            while (it.hasNext() && count++ < limit) {
-                c.add(it.next());
+            List<BlueRun> it = Lists.newArrayList(blueRunContainer.list());
```

Comment:
Why create a new ArrayList that's just thrown away? why not just `allRunList.addAll(blueRunContainer.list())`?

Reply:
blueRunContainer.list() did not actually return List, but returned Iterable